/*
 * FALSA Model Problem
 * 
 * Copyright 2024 Carnegie Mellon University.
 * 
 * NO WARRANTY. THIS CARNEGIE MELLON UNIVERSITY AND SOFTWARE ENGINEERING
 * INSTITUTE MATERIAL IS FURNISHED ON AN "AS-IS" BASIS. CARNEGIE MELLON
 * UNIVERSITY MAKES NO WARRANTIES OF ANY KIND, EITHER EXPRESSED OR IMPLIED, AS
 * TO ANY MATTER INCLUDING, BUT NOT LIMITED TO, WARRANTY OF FITNESS FOR PURPOSE
 * OR MERCHANTABILITY, EXCLUSIVITY, OR RESULTS OBTAINED FROM USE OF THE
 * MATERIAL. CARNEGIE MELLON UNIVERSITY DOES NOT MAKE ANY WARRANTY OF ANY KIND
 * WITH RESPECT TO FREEDOM FROM PATENT, TRADEMARK, OR COPYRIGHT INFRINGEMENT.
 * 
 * Licensed under a MIT (SEI)-style license, please see license.txt or contact
 * permission@sei.cmu.edu for full terms.
 * 
 * [DISTRIBUTION STATEMENT A] This material has been approved for public
 * release and unlimited distribution.  Please see Copyright notice for non-US
 * Government use and distribution.
 * 
 * This Software includes and/or makes use of Third-Party Software each subject
 * to its own license.
 * 
 * DM24-0251
 */


syntax = "proto3";

package uav;

import "Waypoint.proto";

// import "Types.proto";

// https://stackoverflow.com/a/66118518
import "google/protobuf/wrappers.proto";
import "google/protobuf/empty.proto";


service Guidance {

    // void addWaypoint(Waypoint waypoint ) 
    rpc addWaypoint (Waypoint) returns (.google.protobuf.Empty) {}

    // void clearRoute( ) 
    rpc clearRoute (.google.protobuf.Empty) returns (.google.protobuf.Empty) {}

    // int getWaypointCount( ) 
    rpc getWaypointCount (.google.protobuf.Empty) returns (.google.protobuf.Int32Value) {}

    // void land( ) 
    rpc land (.google.protobuf.Empty) returns (.google.protobuf.Empty) {}

    // void returnToBase( ) 
    rpc returnToBase (.google.protobuf.Empty) returns (.google.protobuf.Empty) {}

    // void start( ) 
    rpc start (.google.protobuf.Empty) returns (.google.protobuf.Empty) {}

    // void subscribeStatus(int periodMsec ) 
    rpc subscribeStatus (.google.protobuf.Int32Value) returns (.google.protobuf.Empty) {}

    // void takeoff(double takeoffAltitude ) 
    rpc takeOff (.google.protobuf.DoubleValue) returns (.google.protobuf.Empty) {}

    // void arm( ) 
    rpc arm (.google.protobuf.Empty) returns (.google.protobuf.Empty) {}

    // void disarm( ) 
    rpc disarm (.google.protobuf.Empty) returns (.google.protobuf.Empty) {}
}


